lib/package_provider/cached_package.rb

```diff
@@ -14,52 +14,41 @@ class PackingInProgress < StandardError
     ERROR = '.error'
 
     class << self
-      def from_cache(package_request_or_fingerprint)
-        return nil unless package_ready?(package_request_or_fingerprint)
+      def from_cache(package_hash)
+        return nil unless package_ready?(package_hash)
         Metriks.meter('packageprovider.package.cached').mark
-        path_to_package(package_request_or_fingerprint)
+        path_to_package(package_hash)
       end
 
-      def package_ready?(package_request_or_fingerprint)
-        path = package_path(package_request_or_fingerprint)
+      def package_ready?(package_hash)
+        path = package_path(package_hash)
 
         Dir.exist?(path) && File.exist?(path + PACKAGE_READY) &&
           File.exist?(File.join(path, 'package.zip')) &&
           !File.exist?(path + PACKAGE_CLONE_LOCK)
       end
 
-      def package_path(package_request_or_fingerprint)
-        fp = package_request_fingerprint(package_request_or_fingerprint)
-        File.join(PackageProvider.config.package_cache_root, fp)
+      def package_path(package_hash)
+        File.join(PackageProvider.config.package_cache_root, package_hash)
       end
 
-      def errors(package_request_or_fingerprint)
-        fp = package_request_fingerprint(package_request_or_fingerprint)
-        return unless File.exist?(package_path(fp) + ERROR)
-        File.read(package_path(fp) + ERROR)
+      def errors(package_hash)
+        return unless File.exist?(package_path(package_hash) + ERROR)
+        File.read(package_path(package_hash) + ERROR)
       end
 
       private
 
-      def path_to_package(package_request_or_fingerprint)
-        fp = package_request_fingerprint(package_request_or_fingerprint)
-        File.join(package_path(fp), 'package.zip')
-      end
-
-      def package_request_fingerprint(package_request_or_fingerprint)
-        if package_request_or_fingerprint.respond_to?(:fingerprint)
-          package_request_or_fingerprint.fingerprint
-        else
-          package_request_or_fingerprint
-        end
+      def path_to_package(package_hash)
```

Comment:
Confusing method names for 'package_path' and 'path_to_package' methods. I recommend something like 'package_directory' and 'package_fullpath'.